lib/logthrdestdrv.c

```diff
@@ -255,10 +245,11 @@ _process_result(LogThreadedDestDriver *self, gint result)
 
     case WORKER_INSERT_RESULT_SUCCESS:
       _accept_batch(self);
+      step_sequence_number(&self->seq_num);
```

Comment:
Theoretical question: do we expect the sequence number to be continuous without skips? Should we step down the sequence number in case of drop?

Reply:
No we shouldn't step down the sequence number. The behaviour of SEQNUM is pretty much syslog specific, so might not even matter that much.

The point of SEQNUM is to be able to detect drops and order log messages. I would try to avoid duplicate SEQNUMs at all costs and it is still better to produce gaps between them.

Quoting https://www.cisco.com/c/en/us/td/docs/routers/access/wireless/software/guide/SysMsgLogging.html#wp1054751

> Because there is a chance that more than one log message can have the same timestamp, 
> you can display messages with sequence numbers so that you can unambiguously refer to a > single message. By default, sequence numbers in log messages are not displayed. 

The RFC5424 [meta sequenceId] is structured data is coming from the same roots.

With that said, I simply left SEQNUM always increasing as we do in all other drivers. (e.g. logwriter based ones, and sql).

While researching the subject, I've noticed that logwriter based destinations would use a separate sequence number for local messages and for relayed messages (reproducing the original one the relay has received). Honestly I am a bit confused, which is sad, given that I have written that code, roughly 10 years ago.

What I am sure about is that I don't want to fix SEQNUM logic in this PR. :)

In SQL (which I would like to migrate to this framework in a followup PR), the sequence number if just monotonically increasing, even if the transaction fails is being retried, etc.

So I would just reproduce that logic here and then file an issue to fix this mess up.

Reply:
#2152 is the issue number for the SEQNUM mess.